extensions/blocks/image-compare/view.js

```diff
@@ -0,0 +1,559 @@
+/**
+ * Juxtapose - v1.2.0 - 2017-12-18
+ * Copyright (c) 2017 Alex Duner and Northwestern University Knight Lab
+ *
+ * This code is governed by the terms of the Mozilla Public License (MPL) v2.0,
+ * which is available here: https://github.com/NUKnightLab/juxtapose/blob/master/LICENSE
+ *
+ * This code is incorporated into Image Compare plugin, which is licensed under
+ * GPLv2+, however you may use Juxtapose code separately under the terms of its
+ * original MPL 2.0 license if you wish.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import domReady from '@wordpress/dom-ready';
+
+/**
+ * Internal dependencies
+ */
+import './view.scss';
+
+domReady( function() {
+	const juxtapose = {
+		sliders: [],
+		OPTIMIZATION_ACCEPTED: 1,
+		OPTIMIZATION_WAS_CONSTRAINED: 2,
+	};
+
+	function Graphic( properties, slider ) {
+		const self = this;
+		this.image = new Image();
+
+		this.loaded = false;
+		this.image.onload = function() {
+			self.loaded = true;
+			slider._onLoaded();
+		};
+
+		this.image.src = properties.src;
+		this.image.alt = properties.alt || '';
+		this.label = properties.label || false;
+		this.credit = properties.credit || false;
+	}
+
+	function getImageDimensions( img ) {
+		const dimensions = {
+			width: img.naturalWidth,
+			height: img.naturalHeight,
+			aspect: function() {
+				return this.width / this.height;
+			},
+		};
+		return dimensions;
+	}
+
+	function addClass( element, c ) {
+		if ( element.classList ) {
+			element.classList.add( c );
+		} else {
+			element.className += ' ' + c;
+		}
+	}
```

Comment:
I didn't experience any issues with this in Jetpack, but ran into the following error in the editor, after uploading the first image in the block, when testing D42965-code:

```
Uncaught TypeError: Cannot read property 'classList' of null
```

Reply:
Logged here: #15743